### PR TITLE
added linux compilation support

### PR DESCRIPTION
**Pull request by ko1N**

### CODE REVIEW

**Comment by ko1N** [comment]
please test on windows.

File: NativeCore/ReClassNET_Plugin.hpp [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #pragma once
+#ifndef __RECLASS_NET_PLUGIN_H__
+#define __RECLASS_NET_PLUGIN_H__
```
Comment (KN4CK3R) [inline NativeCore/ReClassNET_Plugin.hpp]: Why is this needed? GCC understands `#pragma once`

File: NativeCore/ReClassNET_Plugin.hpp [hunk 2]
```diff
@@ -7,6 +9,12 @@
 #include <locale>
 #include <cstring>
 
+// OS Specific
+
+#ifdef __linux__
+	#define __stdcall
```
Comment (KN4CK3R) [inline NativeCore/ReClassNET_Plugin.hpp]: Is there a reason for this define? `__stdcall` should not be present in the linux files.

File: ReClass.NET/Memory/ProcessInfo.cs [hunk 3]
```diff
@@ -26,6 +26,11 @@ public ProcessInfo(IntPtr id, string name, string path)
 			{
 				using (var i = NativeMethods.GetIconForFile(Path))
 				{
+					if (i == null)
+					{
+						return null;
+					}
+					
 					return i.ToBitmap();
```
Comment (KN4CK3R) [inline ReClass.NET/Memory/ProcessInfo.cs]: Could you change this to `return i?.ToBitmap();`?

File: ReClass.NET/Memory/ProcessInfo.cs [hunk 4]
```diff
@@ -26,6 +26,11 @@ public ProcessInfo(IntPtr id, string name, string path)
 			{
 				using (var i = NativeMethods.GetIconForFile(Path))
 				{
+					if (i == null)
```
Comment (KN4CK3R) [inline ReClass.NET/Memory/ProcessInfo.cs]: Could you change this to `return i?.ToBitmap();`?

File: ReClass.NET/Plugins/PluginManager.cs [hunk 5]
```diff
@@ -63,10 +65,10 @@ private void LoadPlugins(IEnumerable<FileInfo> files, ILogger logger)
 				{
 					fvi = FileVersionInfo.GetVersionInfo(fi.FullName);
 
-					if (fvi.ProductName != PluginInfo.PluginName && fvi.ProductName != PluginInfo.PluginNativeName)
+					/*if (fvi.ProductName != PluginInfo.PluginName && fvi.ProductName != PluginInfo.PluginNativeName)
```
Comment (KN4CK3R) [inline ReClass.NET/Plugins/PluginManager.cs]: Maybe a parameter `bool checkProductName` should be used for `LoadPlugins` to keep the check for supported files (exe, dll).

File: NativeCore/ReClassNET_Plugin.hpp (hunk 2 above)
Comment (KN4CK3R) [reply NativeCore/ReClassNET_Plugin.hpp]: Ok, this change is because of your changes to the sample plugin. I think it would be better to split the native sample plugin in a Windows and Linux project.

File: NativeCore/ReClassNET_Plugin.hpp (hunk 1 above)
Comment (ko1N) [reply NativeCore/ReClassNET_Plugin.hpp]: its good practice in my opinion since it was windows specific for quite some time

File: NativeCore/ReClassNET_Plugin.hpp (hunk 2 above)
Comment (ko1N) [reply NativeCore/ReClassNET_Plugin.hpp]: __stdcall isnt present in gcc and since linux only really uses stdcall its not needed to supply the calling convention. however we could change it to

	#define __stdcall __attribute__((__stdcall__))

which would result in some compiler warnings

dllmain.cpp:12:86: warning: ‘stdcall’ attribute ignored [-Wattributes]

File: ReClass.NET/Memory/ProcessInfo.cs (hunk 4 above)
Comment (ko1N) [reply ReClass.NET/Memory/ProcessInfo.cs]: im not hugely familiar with c#, does this check if i is null before calling ToBitmap?

File: ReClass.NET/Plugins/PluginManager.cs (hunk 5 above)
Comment (ko1N) [reply ReClass.NET/Plugins/PluginManager.cs]: Yea, the version info isnt present at all under linux which made it skip my native plugins at all. not sure what the best solution would be here

Comment (ko1N) [reply ReClass.NET/Plugins/PluginManager.cs]: we could add the parameter and add a mono specific check to LoadPlugins depending on the OS

File: ReClass.NET/Memory/ProcessInfo.cs (hunk 4 above)
Comment (Timboy67678) [reply ReClass.NET/Memory/ProcessInfo.cs]: yeah its incredibly usefull, https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-conditional-operators

**Comment by ko1N** [comment]
incorporated your change requests
